### PR TITLE
Database

### PR DESCRIPTION
**Pull request by vjt4**
Added back end database models w/ relations

### CODE REVIEW

File: server/db/db.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const Sequelize = require('sequelize')
 const pkg = require('../../package.json')
 
 const databaseName = pkg.name + (process.env.NODE_ENV === 'test' ? '-test' : '')
+console.log(pkg.name)
```
Comment (Kevin-Thai) [inline server/db/db.js]: Remove the console.log

File: server/db/models/cart.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+module.exports = db.define('cart', {
+    quantity: {
+	type: Sequelize.INTEGER,
```
Comment (Kevin-Thai) [inline server/db/models/cart.js]: Fix indentation

File: server/db/models/index.js [hunk 3]
```diff
@@ -1,4 +1,30 @@
...
+//Item User relation through Review
+User.belongsToMany(Item, {through: Review, foreignKey: 'itemId'})
+Item.belongsToMany(User, {through: Review, foreignKey: 'userId'})
+
+Order.belongsTo(User)
+User.hasMany(Order)
+
+Item.belongsTo(Category)
+
+Address.belongsTo(User)
```
Comment (Kevin-Thai) [inline server/db/models/index.js]: User.hasOne(Address)

File: server/db/models/index.js [hunk 4]
```diff
@@ -1,4 +1,30 @@
...
+Item.belongsToMany(Order, {through: OrderItem, foreignKey: 'orderId'})
+
+//Item User relation through Review
+User.belongsToMany(Item, {through: Review, foreignKey: 'itemId'})
+Item.belongsToMany(User, {through: Review, foreignKey: 'userId'})
+
+Order.belongsTo(User)
+User.hasMany(Order)
+
+Item.belongsTo(Category)
```
Comment (Kevin-Thai) [inline server/db/models/index.js]: Category.hasMany(Item)

File: server/db/models/item.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  stock: {
+    type: Sequelize.INTEGER,
+    allowNull: false,
+    validate: {
+      notEmpty: true,
+      min: 0
+    }
+  },
+  description: {
+    type: Sequelize.STRING,
```
Comment (Kevin-Thai) [inline server/db/models/item.js]: Should be Sequelize.TEXT

File: server/db/models/order.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+module.exports = db.define('order', {
+  status: {
+    type: Sequelize.STRING,
+    allowNull: false,
+    validate: {
+      isIn: [['shipped', 'delivered', 'cancelled']]
```
Comment (Kevin-Thai) [inline server/db/models/order.js]: Maybe change 'shipped' to 'processing' or add 'processing' as a separate

Comment (Kevin-Thai) [reply server/db/models/order.js]: Also, set default to 'processing'

File: server/db/models/review.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+module.exports = db.define('review', {
+  rating: {
+    type: Sequelize.BOOLEAN,
+    allowNull: false
+  },
+  description: {
+    type: Sequelize.STRING,
```
Comment (Kevin-Thai) [inline server/db/models/review.js]: Should be Sequelize.TEXT

File: server/db/models/user.js [hunk 8]
```diff
@@ -26,6 +26,24 @@ const User = db.define('user', {
...
+    }
+  },
+  lastName: {
+    type: Sequelize.STRING,
+    allowNull: false,
+    validate: {
+      notEmpty: true
+    }
+  },
+  admin: {
```
Comment (Kevin-Thai) [inline server/db/models/user.js]: Should also not be null or empty

Comment (vjt4) [reply server/db/models/user.js]: If it is null or empty, the default value will set it to false.

**Pull request merged**